ocaml/lang/correctness/physical_vs_structural.yaml

```diff
@@ -0,0 +1,6 @@
+rules:
+- id: physical-not-equal
+  pattern: $X != $Y
+  message: You probably want the structural inequality operator <>.
```

Comment:
I would explain how to use physical equality without triggering a semgrep error.